src/libostree/ostree-repo-pull.c

```diff
@@ -366,6 +391,18 @@ check_outstanding_requests_handle_error (OtPullData          *pull_data,
       if (!pull_data->caught_error)
         {
           pull_data->caught_error = TRUE;
+
+          if (!dir_check_for_free_space (pull_data->repo))
+            {
+              g_autofree char *msg = g_strdup (error->message);
+
+              g_clear_error (errorp);
+              g_set_error (errorp,
+                           G_IO_ERROR,
+                           G_IO_ERROR_NO_SPACE,
+                           "Not enough disk space : %s", msg) ;
```

Comment:
It's weird that we're replacing the error - if I get a 404 because I typo'd an URL but see an error message about disk space that'd be very confusing.   Maybe instead suffix the error?

```
g_autofree char *msg = g_steal_pointer (&error->message);
error->message = g_strconat (msg, " (low disk, cache pruned)", NULL);
```
or so?